src/device/rcp/rsp/rsp_core.h

```diff
@@ -75,6 +75,22 @@ enum sp_registers2
     SP_REGS2_COUNT
 };
 
+enum sp_dma_dir
+{
+    SP_NO_DMA,
```

Comment:
It probably isn't a huge deal, but this is there so that when the struct is initialized by ```memset(sp->fifo, 0, SP_DMA_FIFO_SIZE*sizeof(struct sp_dma));```, the dma_dir is SP_NO_DMA instead of read/write, which in my mind is more correct.

Reply:
Not a huge fan either of SP_NO_DMA as it is "redundant" with the BUSY_FLAG :
- if BUSY flag is cleared, there is no DMA and fifo content has no meaning
- if BUSY flag is set, at least one DMA (or 2 if FULL flag is set) is in the fifo and it is either a READ or a WRITE DMA.